ot2/octopus-full-ssh.py

```diff
@@ -0,0 +1,664 @@
+            # consolidating VE plates 5-8 to Pool Plate columns 5-8
+        if NUM_PLATES >= 5:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('5'))
+        if NUM_PLATES >= 6:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('6'))
+        if NUM_PLATES >= 7:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('7'))
+        if NUM_PLATES == 8:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('8'))
+
+        ### further consolidate plate columns into single wells on the DNA pool deep well plate
+        self.set_flow_rate(48*2, 48, 48)
+        for i in range(NUM_PLATES):
+            self.p300.pick_up_tip(self.tips_200[0].rows()[7-i][NUM_PLATES])
+            # aspirate first four wells
+            for j in range(4):
+                self.p300.aspirate(50, self.consolidation_plate.rows()[j][i].bottom(z=0.5))
+            self.p300.dispense(200, self.pool_plate.rows()[i][0])
+            # aspirate last four wells
+            for j in range(4):
+                self.p300.aspirate(50, self.consolidation_plate.rows()[j+4][i].bottom(z=0.5))
+            self.p300.dispense(200, self.pool_plate.rows()[i][0])
+            self.p300.drop_tip()
+
+        ### add and incubate Purification Beads with pooled samples
+        self.p300.pick_up_tip()
+        self.set_flow_rate(48*4, 48*4, 48)
+        self.p300.mix(10, 200, Purification_Beads.bottom(z=10.0))
+            # adding beads to pool
+        self.set_flow_rate(48*2, 48*4, 48)
+        for i in range(3):
+            self.p300.aspirate(200, Purification_Beads)
+            self.p300.dispense(200, DNA_pool)
+        self.p300.aspirate(120, Purification_Beads)
+        self.p300.dispense(120, DNA_pool)
+            # mix beads well with pool and incubate
+        self.set_flow_rate(48*4, 48*6, 48)
+        self.p300.mix(10, 200, DNA_pool.bottom(z=10.0))
+        self.set_flow_rate(48*4, 48*4, 48)
+        for i in range(20):
+            self.p300.aspirate(200, DNA_pool.bottom(z=10.0))
+            self.p300.dispense(200, DNA_pool.center())
+        self.set_flow_rate(48*4, 48*6, 48/6)
+        self.p300.mix(20, 200, DNA_pool.bottom(z=10.0))
+        self.p300.move_to(DNA_pool.top())
+        self.protocol.delay(seconds=5.0)
+        self.p300.blow_out(DNA_pool.top())
+        self.p300.drop_tip()
+        self.protocol.delay(minutes=10)
+        self.mag_mod.engage()
+        self.protocol.delay(minutes=12)
```

Comment:
Yes, I understand where you are coming from. Many sections throughout the protocol indeed have repeated units of code. 
However, many have intricate differences, such as differences in delay timing and pipette `z` positioning. While these details technically could be passed as arguments within methods, I feel that readability might be more compromised that way with having a wall of out-of-context parameters as opposed to how it is structured currently where I attempt to alleviate the problem by adding line breaks and comments between meaningful operations. This is arguably especially true since these operations are highly localized and don't occur outside of those comment blocks.
Operations not locally unique I have created methods for, such as `set_flow_rate` and `discard_supernatant`.

Reply:
That all sounds a bit hypocritical given that the whole protocol itself is contained within two giant functions `Pool()` and `Post_Pool()`, but I believe having these super high-level sections be separated using massive functions contributes more to readability than having the highly localized low-level sections be their own functions.